graspy/match/faq.py

```diff
@@ -0,0 +1,261 @@
+    
+    perm_inds_ : array, size (n,) where n is the number of vertices in the graphs
+        fitted.
+        The indices of the optimal permutation on the nodes of B, found via
+        FAQ, to best minimize the objective function :math:`f(P) = trace(A^T PBP^T )`.
+
+        
+    score_ : float
+        The objective function value of for the optimal permutation found.
+        
+
+    References
+    ----------
+    .. [1] J. T. Vogelstein, J. M. Conroy, V. Lyzinski, L. J. Podrazik, S. G. Kratzer,
+           E. T. Harley, D. E. Fishkind, R. J. Vogelstein, and C. E. Priebe, “Fast
+           approximate quadratic programming for graph matching,” PLOS one, vol. 10,
+           no. 4, p. e0121002, 2015.
+
+    """
+
+    def __init__(
+        self,
+        n_init=1,
+        init_method="barycenter",
+        max_iter=30,
+        shuffle_input=True,
+        eps=0.1,
+        gmp=False,
+    ):
+
+        if n_init > 0 and type(n_init) is int:
+            self.n_init = n_init
+        else:
+            msg = '"n_init" must be a positive integer'
+            raise TypeError(msg)
+        if init_method == "rand":
+            self.init_method = "rand"
+        elif init_method == "barycenter":
+            self.init_method = "barycenter"
+            self.n_init = 1
+        else:
+            msg = 'Invalid "init_method" parameter string'
+            raise ValueError(msg)
+        if max_iter > 0 and type(max_iter) is int:
+            self.max_iter = max_iter
+        else:
+            msg = '"max_iter" must be a positive integer'
+            raise TypeError(msg)
+        if type(shuffle_input) is bool:
+            self.shuffle_input = shuffle_input
+        else:
+            msg = '"shuffle_input" must be a boolean'
+            raise TypeError(msg)
+        if eps > 0 and type(eps) is float:
+            self.eps = eps
+        else:
+            msg = '"eps" must be a positive float'
+            raise TypeError(msg)
+        if type(gmp) is bool:
+            self.gmp = gmp
+        else:
+            msg = '"gmp" must be a boolean'
+            raise TypeError(msg)
+
+    def fit(self, A, B):
+        """
+        Fits the model with two assigned adjacency matrices
+        
+        Parameters
+        ---------
+        A : 2d-array, square, positive
+            A square, positive adjacency matrix
+            
+        B : 2d-array, square, positive
+            A square, positive adjacency matrix
+
+        Returns
+        -------
+        
+        self : returns an instance of self
+        """
+
+        A = check_array(A, copy=True, ensure_2d=True)
+        B = check_array(B, copy=True, ensure_2d=True)
+        n = A.shape[0]  # number of vertices in graphs
+
+        if self.shuffle_input:
+            node_shuffle_input = np.random.permutation(n)
+            A = A[np.ix_(node_shuffle_input, node_shuffle_input)]
+            # shuffle_input to avoid results from inputs that were already matched
+        min = 1
```

Comment:
`min` is a python keyword, unfortunately, so it's bad practice to use as a var name. `obj_func_scalar`? Or whatever else you think of is probably fine!

Reply:
@asaadeldin11